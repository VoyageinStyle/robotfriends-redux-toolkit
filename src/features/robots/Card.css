.cardInfo h2,
.cardInfo p {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.hidden {
  display: none !important;
}