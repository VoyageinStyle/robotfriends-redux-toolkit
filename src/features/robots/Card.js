import React from "react";

const Card = ({ id, name, email, selected, hidden }) => {
  const isHidden = () => (hidden === null || hidden === id ? "" : "hidden");
  const onClick = () => {
    return selected(id);
  };

  return (
    <div
      className={`bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5 card ${isHidden()}`}
      onClick={onClick}
    >
      <img
        alt={`robot-${id}`}
        src={`https://robohash.org/${id}?size=200x200`}
        id={id}
        loading="lazy"
        height="200px"
        width="200px"
      />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;