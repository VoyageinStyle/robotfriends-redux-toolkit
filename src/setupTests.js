import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import { setupListeners } from "@reduxjs/toolkit/dist/query";

import fetchMock from "jest-fetch-mock";

import { robotsApi } from "./services/robotsAPI";
import counterReducer from "./features/counter/counterSlice";
import searchRobotsReducer from "./features/searchBox/searchBoxSlice";

import configureMockStore from "redux-mock-store";
const setupStore = configureMockStore();
export const mockStore = setupStore();

fetchMock.enableMocks();

export const renderWithProviders = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: {
        counter: counterReducer,
        searchRobots: searchRobotsReducer,
        [robotsApi.reducerPath]: robotsApi.reducer,
      },
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware().concat(robotsApi.middleware),
      preloadedState,
    }),
    ...renderOptions
  } = {}
) => {
  setupListeners(setupStore.dispatch);
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
};
