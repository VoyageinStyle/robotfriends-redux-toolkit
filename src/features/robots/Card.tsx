import React from "react";
import "./Card.css";
import CardDetails from "./CardDetails";
import { IRobot } from "./CardList";

interface ICardProps {
  id: number;
  robot: IRobot;
  selected: (id: number) => number | null;
  hidden: number | null;
}

const Card = ({ id, robot, selected, hidden }: ICardProps) => {
  const isHidden = () => (hidden === null || hidden === id ? "" : "hidden");
  const onClick = () => () => selected(id);
  const lazyOrAuto = (id: number) => (id > 6 ? "lazy" : "eager");
  const width = () => (hidden ? "" : "w5");

  return (
    <div
      className={`bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5 card ${width()} ${isHidden()}`}
      onClick={onClick()}
    >
      <img
        alt={`robot-${id}`}
        src={`https://robohash.org/${id}?size=200x200`}
        id={id.toString()}
        loading={lazyOrAuto(id)}
        height={"200px"}
        width={"200px"}
      />
      <div className="cardInfo" id="cardInfo">
        <h2>{robot.name}</h2>
        <p>{robot.email}</p>
      </div>
      <CardDetails hidden={hidden} robot={robot} />
    </div>
  );
};

export default Card;
