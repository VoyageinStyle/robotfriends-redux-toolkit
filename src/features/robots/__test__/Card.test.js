import React from "react";
import { screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import Card from "../Card";
import { renderWithProviders } from "../../../setupTests";

const mockCardDetails = jest.fn();
jest.mock("../CardDetails", () => () => {
  mockCardDetails();
  return <modal-carddetails />;
});

const cardProps = {
  robot: {
    name: "Jeanne Graham",
    id: 1,
    email: "Sincere@april.biz",
  },
};

it("receives props", () => {
  renderWithProviders(<Card {...cardProps} />);
  const details = screen.getByText(/jeanne/i);
  expect(details).toBeInTheDocument();
});
it("hides other cards when clicked", () => {
  renderWithProviders(<Card {...cardProps} />);
  const details = screen.getByText(/jean/i);
  expect(details).toMatchSnapshot()
});
