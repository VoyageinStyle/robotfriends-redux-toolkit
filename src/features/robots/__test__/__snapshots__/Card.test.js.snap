// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides other cards when clicked 1`] = `
<h2>
  Jeanne Graham
</h2>
`;
